vast/data_prep/readHDF5.py

```diff
@@ -57,7 +57,9 @@ def read_features(args, feature_file_names=None, cls_to_process=None):
                 image_names = hf[cls]["image_names"][()].tolist()
             else:
                 image_names = None
-            features = torch.cat(temp, dim=1)
+            if len(temp[0].shape) == 1:
+                temp[0] = temp[0][None,:]
```

Comment:
@RRabinow Thanks for pointing this out and proposing a fix.
Looks like you are only checking and fixing for the zeroth index of the `temp` list, but there is no guarantee that the `temp` list will always have a length of `1` since in line 55 `temp` is being appended in a for loop. If your test cases were only using one feature file, yes your fix will work but it will break the case where features from multiple files need to be concatenated.
So the **solution** would be either to run your check and fix in a for loop running for all elements of temp or a better alternative would be to store the value being appended into temp at line 55 in a temporary variable, then perform your check and fix on that variable, which will then will be appended into the `temp` list.
Let me know if this makes sense.

Reply:
@akshay-raj-dhamija I did not consider multiple feature files being read. I will adjust the code to apply the fix on line 55 instead. Thank you.